### DISCUSSION TITLE
QZ settings for using Kinomap with Echelon for auto resistance

### DISCUSSION THREAD

**Discussion by Sl0wRider** [cagnulein/qdomyos-zwift#451] [Q&A]
Would it be possible to post the standard settings to use when connecting an Echelon spin bike to Kinomap using the QZ app to ensure for accurate automatic program resistance adjustments?

**Comment by cagnulein** [comment]
Hi! First of all are you seeing the QZ device correctly in kinomap? If so i
guess you should:
1) enable the zwift erg mode
2) set 10 to the zwift erg filter up and down
3) set 16 to zwift resistance offset

Let me know if you need more assistance

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 13 ott 2021 alle ore 10:25 Kinomap ***@***.***>
ha scritto:

**Reply by Sl0wRider** [reply]
Hey Roberto, so as we discovered before, the settings in the QZ app are not the same for Kinomap as they are for Zwift. After a ton of trial and error I have found QZ app settings that should make Kinomap work for most if not all people (At least for Echelon users, but hopefully others too), so please post this on your "How To" web page so that others, especially Echelon Users, will be able to benefit from the results, and again, as always, a major thanks to no end for creating this app for all of us to enjoy our fitness routines all the more!!! 

First, when using Kinomap, the user needs to make sure that in the "Training" settings, both "Rate of Resistance Change" and "Maximum Slope" have to be maxed at at 15% otherwise the following settings will not work.

In the QZ app in "Bike Settings"you have to TURN OFF "Zwift Workout/Erg Mode" otherwise the next following Bike Settings will not work.

There are two settings profiles for "Video"  Virtual Rides and for "Structured Workouts"

QZ app "Bike Settings" for "Video" Virtual Rides: Along with Zwift Workout/Erg Mode turned off, you need to set the "Zwift Resistance Offset"  to "16"  and the "Zwift  Resistance Gain" to "0.65 ". The other options of " Zwift Erg Up Filter" and Zwift Erg Down Filter" can be both set as 10, just like for Zwift, but I'm not sure that it impacts anything. The final setting of "Resistance at Start-up"  is whatever resistance level the user wants to start with, but it will have no bearing on the actual exercise program. 

QZ app "Bike Settings" for "Structured Workouts" : Along with Zwift Workout/Erg Mode turned off, you need to set the "Zwift Resistance Offset"  to "0" and the "Zwift  Resistance Gain" to "0.08" . The other options of " Zwift Erg Up Filter" and Zwift Erg Down Filter" can be both set as 10, just like for Zwift, but I'm not sure that it impacts anything. The final setting of "Resistance at Start-up"  is whatever resistance level the user wants to start with, but it will have no bearing on the actual exercise program.

Along with those QZ app settings for "Structured Workouts"  the user can only choose "Resistance" programs in Kinomap and NOT FTP, WATT or SLOPE for the programs to automatically change the resistance properly.

Again, these settings definitely work for Echelon users with a max of 32 resistance, but hopefully they'll work for others. If not, I'm sure a couple slight changes will make it work for others too.

I will email this info to you too with the QZ app settings that I use.

**Reply by cagnulein** [reply]
added in the wiki! Thanks @Sl0wRider !

**Comment by Sl0wRider** [comment]
Yes, I can access the QZ app properly in Kinomap. I attempted the settings that you suggested and with Zwift Erg Mode selected the device makes no attempt to adjust resistance automatically. When Zwift Erg Mode is not selected the QZ app does receive automated resistance change responses, but the other settings must not be accurate because the resistance started at 1 and after a small program resistance increase, the resistance changed to 32.Thanks,TomOn Oct 13, 2021 04:38, Roberto Viola ***@***.***> wrote:

Hi! First of all are you seeing the QZ device correctly in kinomap? If so i

guess you should:

1) enable the zwift erg mode

2) set 10 to the zwift erg filter up and down

3) set 16 to zwift resistance offset



Let me know if you need more assistance



Roberto Viola

Software engineer and open source enthusiast

http://robertoviola.cloud







Il giorno mer 13 ott 2021 alle ore 10:25 Kinomap ***@***.***>

ha scritto:

**Comment by cagnulein** [comment]
Ok, in order to understand the situation, send me the debug log from QZ collected from a kinomap session and i will help you to to understand what's happening

**Comment by Sl0wRider** [comment]
I received a "great workout" email from QZ with graphs and a .fit file, but no debug log. I ran a search on my phone for keywords debug and QZ but couldn't find it. Do you know what folder they can be found in an Android Samsung? I did have the Debug Log option selected as "On" in the settings by the way.On Oct 13, 2021 05:24, Roberto Viola ***@***.***> wrote:
Ok, in order to understand the situation, send me the debug log from QZ collected from a kinomap session and i will help you to to understand what's happening

**Comment by cagnulein** [comment]
you have to use the free app CX file explorer on android and go into the /android/data/org.cagnulein.qdomyos folder

**Comment by Sl0wRider** [comment]
Good to go, thanks. I believe these are the correct logs.On Oct 13, 2021 05:59, Roberto Viola ***@***.***> wrote:
you have to use the free app CX file explorer on android and go into the /android/data/org.cagnulein.qdomyos folder

**Comment by cagnulein** [comment]
sorry i didn't see the attachments

**Comment by Sl0wRider** [comment]
Trying again.On Oct 13, 2021 08:18, Roberto Viola ***@***.***> wrote:
sorry i didn't see the attachments

**Reply by cagnulein** [reply]
you can't add attachment from your email client, you have to do in the github page directly :)

**Comment by Sl0wRider** [comment]
[debug-Wed_Oct_13_05_37_26_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7346601/debug-Wed_Oct_13_05_37_26_2021.log)
[debug-Wed_Oct_13_05_40_25_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7346602/debug-Wed_Oct_13_05_40_25_2021.log)

**Comment by Sl0wRider** [comment]
Hopefully that worked 🤞

**Reply by cagnulein** [reply]
yes it does now :)

**Comment by cagnulein** [comment]
@Kinomap unfortunately there is no sessions at all on the logs. did you do a kinomap session during the log?

**Comment by Sl0wRider** [comment]
Yeah, I'll try the tests again and repost the new debug logs.

**Reply by cagnulein** [reply]
ok thanks, maybe you just picked up the wrong files in the folder

**Comment by Sl0wRider** [comment]
[debug-Thu_Oct_14_10_12_18_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7346803/debug-Thu_Oct_14_10_12_18_2021.log)
[debug-Thu_Oct_14_10_19_12_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7346804/debug-Thu_Oct_14_10_19_12_2021.log)

**Comment by Sl0wRider** [comment]
Those are two five minute segments from the same training profile.

**Comment by Sl0wRider** [comment]
![20211014_102543](https://user-images.githubusercontent.com/92429434/137338360-49a7dccc-3316-429d-96e8-40f2ac9d8960.jpg)

**Comment by Sl0wRider** [comment]
The first five minute segment using the Zwift Erg Mode selected made no efforts for automated resistance, but when I tried again with Zwift Erg Mode not selected you can tell that it tried to make resistance adjustments, it just doesn't know how to do it the correct way.

The four peaks in the image provided shows that QZ attempted to adjust the resistance from 1 to 32, but then immediately went back down to resistance level 1. During the 4th minute the resistance went up to level 32 and held there.

**Comment by cagnulein** [comment]
ok i checked the log. Kinomap did a really strange implementation of FTMS ERG mode. You can try to search in the log the keyword "new requested resistance zwift" to have a look at it yourself.

Just to explain in an easy way: Zwift usually sends ERG mode with a request of wattage. Clear and easy.
Kinomap, instead, send a "virtual inclination" that it's relative to your effort. This is a way to do an easy thing in a hard way!

Probably the only way to manage it, would be:
- adding a new Kinomap parameter in QZ to enable this handling
- tread the kinomap inclination as a relative increment of the resistance. For example 1% grade could be a +1 level of resistance and so on..

**Comment by Sl0wRider** [comment]
Hmmm, I'm not sure if you have ever used Kinomap, but it has two main options, Virtual Rides from videos and Structured Workouts. Virtual rides is your standard elevation to resistance adjustments, but I don't know what methods they use. Structured Workouts follow a set resistance adjustment program, but the graph for target watts does change based on the user's level of effort. So the harder the user works, the graph for target Watts also adjusts,  essentially asking the user to work harder to keep pace with the program. It's an interesting concept that appears to be tailored to each user. i.e. beginners, intermediate and advanced riders should all have similar experiences using the same workout program.On Oct 14, 2021 10:47, Roberto Viola ***@***.***> wrote:
ok i checked the log. Kinomap did a really strange implementation of FTMS ERG mode. You can try to search in the log the keyword "new requested resistance zwift" to have a look at it yourself.
Just to explain in an easy way: Zwift usually sends ERG mode with a request of wattage. Clear and easy.
Kinomap, instead, send a "virtual inclination" that it's relative to your effort. This is a way to do an easy thing in a hard way!
Probably the only way to manage it, would be:
adding a new Kinomap parameter in QZ to enable this handlingtread the kinomap inclination as a relative increment of the resistance. For example 1% grade could be a +1 level of resistance and so on..

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.


On Oct 14, 2021 10:47, Roberto Viola ***@***.***> wrote:
ok i checked the log. Kinomap did a really strange implementation of FTMS ERG mode. You can try to search in the log the keyword "new requested resistance zwift" to have a look at it yourself.
Just to explain in an easy way: Zwift usually sends ERG mode with a request of wattage. Clear and easy.
Kinomap, instead, send a "virtual inclination" that it's relative to your effort. This is a way to do an easy thing in a hard way!
Probably the only way to manage it, would be:
adding a new Kinomap parameter in QZ to enable this handlingtread the kinomap inclination as a relative increment of the resistance. For example 1% grade could be a +1 level of resistance and so on..

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.


On Oct 14, 2021 10:47, Roberto Viola ***@***.***> wrote:
ok i checked the log. Kinomap did a really strange implementation of FTMS ERG mode. You can try to search in the log the keyword "new requested resistance zwift" to have a look at it yourself.
Just to explain in an easy way: Zwift usually sends ERG mode with a request of wattage. Clear and easy.
Kinomap, instead, send a "virtual inclination" that it's relative to your effort. This is a way to do an easy thing in a hard way!
Probably the only way to manage it, would be:
adding a new Kinomap parameter in QZ to enable this handlingtread the kinomap inclination as a relative increment of the resistance. For example 1% grade could be a +1 level of resistance and so on..

**Comment by Sl0wRider** [comment]
Any ideas for a way forward to help bring Kinomap to life with QZ? It sounds complicated. If you would like to use my Kinomap account to troubleshoot I'd be happy to help.

**Reply by cagnulein** [reply]
the issue is that the FTMS value that kinomap sends doesn't have sense at all.. i mean that values should be a incline grade but they are out of scale and i don't know how to treat them. maybe i can use the first one as a offset, and try to use the next ones as a variation..but it's just an idea.

has kinomap a manual or something describes how this work on low level?

**Comment by Sl0wRider** [comment]
No, not that I am aware of :(On Oct 15, 2021 03:55, Roberto Viola ***@***.***> wrote:
the issue is that the FTMS value that kinomap sends doesn't have sense at all.. i mean that values should be a incline grade but they are out of scale and i don't know how to treat them. maybe i can use the first one as a offset, and try to use the next ones as a variation..but it's just an idea.
has kinomap a manual or something describes how this work on low level?
